### PR TITLE
[pr] Fix solana-runtime

### PR DESCRIPTION
**Pull request by DudessaPr**
#### Summary of Changes

Alleviating complications with importing solana-runtime into svm-rollup

Fixes #

### CODE REVIEW

File: accounts-db/src/tiered_storage/hot.rs
Comment (neutrinoks) [inline accounts-db/src/tiered_storage/hot.rs:1]: Still weird but if the compiler wants it, so ok... 🤷😄

**Review by neutrinoks — APPROVED** [review]
Looks very good and clean to me 🚀

File: account-decoder/src/lib.rs [hunk 1]
```diff
@@ -26,10 +26,7 @@ use {
         fee_calculator::FeeCalculator,
         pubkey::Pubkey,
     },
-    std::{
-        io::{Read, Write},
```
Comment (Yiwen-Gao) [inline account-decoder/src/lib.rs:30]: nit: can we comment this out instead of deleting it so that we know what our changes are when looking back? :pray:

File: runtime/src/bank/partitioned_epoch_rewards/calculation.rs [hunk 2]
```diff
@@ -106,10 +106,10 @@ impl Bank {
             stake_rewards_by_partition,
             old_vote_balance_and_staked,
             validator_rewards,
-            validator_rate,
-            foundation_rate,
-            prev_epoch_duration_in_years,
-            capitalization,
+            validator_rate: _,
```
Comment (Yiwen-Gao) [inline runtime/src/bank/partitioned_epoch_rewards/calculation.rs:109]: Why is this needed? For types?

File: runtime/src/bank.rs [hunk 3]
```diff
@@ -2181,9 +2183,9 @@ impl Bank {
         let capitalization = self.capitalization();
         let PrevEpochInflationRewards {
             validator_rewards,
-            prev_epoch_duration_in_years,
-            validator_rate,
-            foundation_rate,
+            prev_epoch_duration_in_years: _,
```
Comment (Yiwen-Gao) [inline runtime/src/bank.rs:2186]: Same question here.

File: runtime/src/bank.rs [hunk 4]
```diff
@@ -6792,12 +6794,13 @@ impl Bank {
         //  1. Transaction forwarding delay
         //  2. The slot at which the next leader will actually process the transaction
         // Drop the transaction if it will expire by the time the next node receives and processes it
-        let api = perf_libs::api();
-        let max_tx_fwd_delay = if api.is_none() {
-            MAX_TRANSACTION_FORWARDING_DELAY
-        } else {
-            MAX_TRANSACTION_FORWARDING_DELAY_GPU
-        };
+        // let api = perf_libs::api();
```
Comment (Yiwen-Gao) [inline runtime/src/bank.rs:6797]: Is `perf_libs` from the `solana-performance` crate?

**Review by Yiwen-Gao — APPROVED** [review]
I left a couple of questions, but the PR looks good to me overall. Thanks!

**Comment by Yiwen-Gao** [comment]
@DudessaPr can you squash [d1ec319](https://github.com/nitro-svm/agave/pull/2/commits/d1ec319e359da6ec2cec0b74f8397211cc96ab36) (apply risc0 changes) and [f48695c](https://github.com/nitro-svm/agave/pull/2/commits/f48695c8e43855f8f9191cb4e1b39ef011a83088) (address warnings) into a single commit to clean up the history on this branch?

Then when you merge this into `feat/risc0`, can you select "Rebase and merge" instead of "Squash and merge" so that we maintain each commit as a separate changeset? Thanks!

File: runtime/src/bank/partitioned_epoch_rewards/calculation.rs (hunk 2 above)
Comment (DudessaPr) [reply runtime/src/bank/partitioned_epoch_rewards/calculation.rs:109]: That is ignored variables. We can put :_ to ignore them as their unused in the function. Simply it is to avoid complains from cargo clippy

File: runtime/src/bank.rs (hunk 4 above)
Comment (DudessaPr) [reply runtime/src/bank.rs:6797]: Yes, you are right

File: runtime/src/bank/partitioned_epoch_rewards/calculation.rs (hunk 2 above)
Comment (DudessaPr) [reply runtime/src/bank/partitioned_epoch_rewards/calculation.rs:109]: These variables are not used. They use interesting property of Rust, they return a struct and instead of creating an object corresponding to that struct they declare its fields. I used ":_" in order to avoid cargo clippy complains about unused variables.

for instance
```
warning: unused variable: `validator_rate`
   --> runtime/src/bank/partitioned_epoch_rewards/calculation.rs:109:13
    |
109 |             validator_rate,
    |             ^^^^^^^^^^^^^^ help: try ignoring the field: `validator_rate: _`
    |
    = note: `#[warn(unused_variables)]` on by default
    ```

Comment (Yiwen-Gao) [reply runtime/src/bank/partitioned_epoch_rewards/calculation.rs:109]: Got it, thanks for the context.
